master/buildbot/status/build.py

```diff
@@ -520,6 +520,18 @@ def checkLogfiles(self):
         for s in self.steps:
             s.checkLogfiles()
 
+    @defer.inlineCallbacks
+    def stopBuild(self, reason, result=None):
+        c = interfaces.IControl(self.master)
+        buildername = self.getBuilder().getName()
+        bldrc = c.getBuilder(buildername)
+        if bldrc:
+            bldc = bldrc.getBuild(self.getNumber())
+            if bldc:
+                yield bldc.stopBuild(reason=reason, result=result)
```

Comment:
@BarryUnity I didn't change variable names, it was before like this I just move the code to common place